vicodyn/include/cocaine/vicodyn/peer.hpp

```diff
@@ -82,15 +83,28 @@ class peer_t : public std::enable_shared_from_this<peer_t> {
 // thread safe wrapper on map of peers indexed by uuid
 class peers_t {
 public:
+    using clock_t = std::chrono::steady_clock;
+
     class app_service_t {
-        using clock_t = std::chrono::steady_clock;
+        using timing_t = std::pair<clock_t::time_point, std::chrono::milliseconds>;
 
         clock_t::time_point ban_until_;
+        std::queue<timing_t> timings_;
```

Comment:
Please, reuse code for moving average from `metrics`: https://github.com/interiorem/metrics/blob/master/include/metrics/usts/ewma.hpp

Reply:
Fixed